### ISSUE TITLE
Add IElementFactory for lists with special behavior

### ISSUE THREAD

**Issue by rmadsen-ks** [opentap/opentap#1798]
Sweep Parameter Step has a SweepRowCollection type which contains SweepRows. That is done because SweepRow needs a reference to the test step it belongs to. 

However, it does not know about the step until it is added to the collection, which makes it hard to figure out which columns SweepRows has when generating a GUI. It would be simpler if there was an API on the test step or list, which tells OpenTAP how to make an instance of the elements, given its context.

It could look like this:

```cs

    /// <summary>
    /// Defines how the class can create list elements of a given type Or for a given member.
    /// If its the class having the list property, elementContext will be a IMemberData
    /// If its the list containing the elements, it will be an ITypeData.
    /// </summary>
    public interface IElementFactory
    {
        /// <summary> Creates a new element for the list. </summary>
        object NewElement(IReflectionData elementContext);
    }
```

**Comment by rmadsen-ks** [comment]
### Testing

- Create a Sweep Parameter step, set it up in the usual way with child step parameterize a property...
- Verify that you *dont* need to add a row for the columns to show up in the sweep values dialog. (In previous version you had to add a row for OpenTAP to be able to discover the column types.)

**Comment by sebastian-pop** [comment]
Reproduced on 9.26 and fixed on 9.27

![Image](https://github.com/user-attachments/assets/4fc0ecfc-ac5c-4655-b58a-ab92c49e71ed)

**Comment by tudor9507** [comment]
https://mikkeller.testrail.com/index.php?/cases/view/21106 
step 3 added to this TC
